timelock-impl/src/main/java/com/palantir/atlasdb/timelock/auth/impl/SimpleAuthorizer.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.auth.impl;
+
+import java.util.Map;
+import java.util.Set;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.palantir.lock.TimelockNamespace;
+import com.palantir.atlasdb.timelock.auth.api.Authorizer;
+import com.palantir.atlasdb.timelock.auth.api.NamespaceMatcher;
+import com.palantir.atlasdb.timelock.auth.api.Client;
+
+public class SimpleAuthorizer implements Authorizer {
+    private final Map<Client, NamespaceMatcher> privileges;
+    private final Set<Client> admins;
+    private final AuthRequirer authRequirer;
+
+    SimpleAuthorizer(Map<Client, NamespaceMatcher> privileges, Set<Client> admins, AuthRequirer authRequirer) {
```

Comment:
If I've understood correctly, we have:
* `Client` which consists of a username and password.
* We have a `PrivilegeConfig` block, which looks roughly like this:
```
privileges:
  - id: "my-first-timelock-client"
    namespaces:
      - namespace_1
      - namespace_2
    namespace-regexes/prefixes:
      - *.*
  - id: "my-admin-client"
    type: admin
```

So effectively from config you get a `List<PrivilegeConfig>` -> `Map<ClientId, Privileges>` where `Privileges` is either `StandardPrivileges` or `AdminPrivileges`. So it's somewhat similar to `NamespaceMatcher` but slightly different. It does mean we can get rid of `Set<Client>` as admins, since you can just call `Privilege.hasPrivilege(Namespace)`.

`AuthRequirer` seems a bit odd, it might make sense to have a class/interface called `Namespaces` and it's that which you ask `isNamespaceLocked`? It takes a `List<StandardPrivileges>`. I imagine this is 0.x only for namespace locking and not anything more general.

With above, it then feels weird to have a `Namespaces`/`AuthRequirer` when `List<PrivilegeConfig>` is the canonical source of truth. So I would just not inject it as a dependency. 🤷‍♂ 

It's unclear to me how if given a config that looks like the above, whether or not it maps to this authoriser thing cleanly imo.

Reply:
I agree; the possibility of inconsistency also bothered me. Though, thinking again; we need to be able to configure Authorizer for different auth requirement cases. So the new implementation takes `Map<Client, Privileges> privileges, AuthRequirement authRequirement`; where `AuthRequirement` is just a simple enum class without any implementation; and that prevents inconsistency I believe.

Reply:
what are the different cases?